### PR TITLE
DRIVERS-2679: make delimiting slash between hosts and connection options optional

### PR DESCRIPTION
**Pull request by W-A-James**
- Change spec language to make delimiting slash between hosts and connection options optional.
- Updated connection-string spec tests to allow optional delimiting slash

Please complete the following before merging:

- [x] Update changelog.
- [x] Make sure there are generated JSON files from the YAML test files.
- [x] Test changes in at least one language driver.
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless).

### CODE REVIEW

File: source/connection-string/tests/valid-options.yml [hunk 1]
```diff
@@ -15,3 +15,11 @@ tests:
...
         options:
             authmechanism: "MONGODB-CR"
+    -
+        description: "Missing delimiting slash between hosts and options"
+        uri: "mongodb://example.com?w=1"
+        valid: true
+        warning: ~
+        hosts: ~
+        auth: ~
+        options: ~
```
Comment (jyemin) [inline source/connection-string/tests/valid-options.yml:25]: Consider adding an assertion that options are respected when there is no delimiting slash (e.g. that the write concern is set here, or some other option if it's easier).

Comment (W-A-James) [reply source/connection-string/tests/valid-options.yml]: Changed this to check for tls being set to true.

File: source/connection-string/tests/valid-options.yml [hunk 2]
```diff
@@ -15,3 +15,12 @@ tests:
             db: "admin"
         options:
             authmechanism: "MONGODB-CR"
+    -
+        description: "Missing delimiting slash between hosts and options"
+        uri: "mongodb://example.com?tls=true"
+        valid: true
+        warning: false
+        hosts: ~
```
Comment (alcaeus) [inline source/connection-string/tests/valid-options.yml:23]: This should also assert that the hostname is parsed correctly:

```suggestion
        hosts:
            -
                type: "hostname"
                host: "example.com"
                port: ~
```

**Comment by W-A-James** [comment]
@jyemin tests have been verified to pass on C# by @BorisDog

File: source/connection-string/tests/valid-options.json [hunk 3]
```diff
@@ -20,6 +20,17 @@
       "options": {
         "authmechanism": "MONGODB-CR"
       }
+    },
+    {
+      "description": "Missing delimiting slash between hosts and options",
+      "uri": "mongodb://example.com?tls=true",
+      "valid": true,
+      "warning": false,
+      "hosts": null,
```
Comment (BorisDog) [inline source/connection-string/tests/valid-options.json:29]: Looks like `hosts` property is different from one yml.

**Review by BorisDog — COMMENTED** [review]
The

File: source/connection-string/tests/valid-options.json (hunk 3 above)
Comment (W-A-James) [reply source/connection-string/tests/valid-options.json]: Good catch, forgot to regen the json files

**Review by BorisDog — APPROVED** [review]
Test passes in C#

**Pull request merged**